opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/frontend/AccountFragment.java

```diff
@@ -847,76 +748,59 @@ public void onClick(View clicked) {
-                        holder.tvFormatDetail.setText(
-                                Html.fromHtml(item.get(AccountData.KEY_LENT_FORMAT)));
-                    } else {
-                        holder.tvFormatDetail.setVisibility(View.GONE);
-                    }
-                    if (item.containsKey(AccountData.KEY_LENT_LENDING_BRANCH)) {
-                        holder.tvBranchDetail
-                                .setText(Html.fromHtml(item.get(AccountData.KEY_LENT_LENDING_BRANCH)));
-                    } else if (item.containsKey(AccountData.KEY_LENT_BRANCH)) {
-                        holder.tvBranchDetail.setText(Html.fromHtml(item.get(AccountData.KEY_LENT_BRANCH)));
+                    setTextOrHide(item.getAuthor(), holder.tvAuthorDetail);
+                    setHtmlTextOrHide(item.getFormat(), holder.tvFormatDetail);
+                    if (item.getLendingBranch() != null && item.getHomeBranch() != null) {
+                        holder.tvBranchDetail.setText(Html.fromHtml(
+                                item.getLendingBranch() + " / " + item.getHomeBranch()));
+                    } else if (item.getLendingBranch() != null) {
+                        holder.tvBranchDetail.setText(Html.fromHtml(item.getLendingBranch()));
                     } else {
-                        holder.tvBranchDetail.setVisibility(View.GONE);
+                        setHtmlTextOrHide(item.getHomeBranch(), holder.tvBranchDetail);
                     }
 
                     // Color codes for return dates
-                    if (item.containsKey(AccountData.KEY_LENT_DEADLINE_TIMESTAMP)) {
-                        if (Long.parseLong(item
-                                .get(AccountData.KEY_LENT_DEADLINE_TIMESTAMP)) < System
-                                .currentTimeMillis()) {
-                            holder.vStatusColor
-                                    .setBackgroundColor(getResources().getColor(R.color.date_overdue));
-                        } else if ((Long.parseLong(item
-                                .get(AccountData.KEY_LENT_DEADLINE_TIMESTAMP)) - System
-                                .currentTimeMillis()) <= tolerance) {
-                            holder.vStatusColor
-                                    .setBackgroundColor(getResources().getColor(R.color.date_warning));
-                        } else if (item.containsKey(AccountData.KEY_LENT_DOWNLOAD)) {
+                    if (item.getDeadline() != null) {
+                        if (item.getDeadline().equals(LocalDate.now()) ||
+                                item.getDeadline().isBefore(LocalDate.now())) {
+                            holder.vStatusColor.setBackgroundColor(
+                                    getResources().getColor(R.color.date_overdue));
+                        } else if (
+                                Days.daysBetween(item.getDeadline(), LocalDate.now()).getDays() <=
```

Comment:
Fixed in 99ea18e. The order of the arguments supplied to `Days.daysBetween` was incorrect.